homeassistant/components/reolink/host.py

```diff
@@ -0,0 +1,176 @@
+"""This component encapsulates the NVR/camera API and subscription."""
+from __future__ import annotations
+
+import logging
+import ssl
+
+import aiohttp
+from reolink_ip.api import Host
+
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import CONF_PROTOCOL, CONF_USE_HTTPS, DEFAULT_PROTOCOL, DEFAULT_TIMEOUT
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ReolinkHost:
+    """The implementation of the Reolink Host class."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config: dict,
+        options: dict,
+    ) -> None:
+        """Initialize Reolink Host. Could be either NVR, or Camera."""
+        self._hass: HomeAssistant = hass
+
+        self._clientsession: aiohttp.ClientSession | None = None
+        self._unique_id: str | None = None
+
+        cur_protocol = (
+            DEFAULT_PROTOCOL if CONF_PROTOCOL not in options else options[CONF_PROTOCOL]
+        )
+
+        self._api = Host(
+            config[CONF_HOST],
+            config[CONF_USERNAME],
+            config[CONF_PASSWORD],
+            port=config.get(CONF_PORT),
+            use_https=config.get(CONF_USE_HTTPS),
+            protocol=cur_protocol,
+            timeout=DEFAULT_TIMEOUT,
+            aiohttp_get_session_callback=self.get_iohttp_session,
+        )
+
+    @property
+    def unique_id(self):
+        """Create the unique ID, base for all entities."""
+        return self._unique_id
+
+    @property
+    def api(self):
+        """Return the API object."""
+        return self._api
+
+    async def init(self) -> bool:
+        """Connect to Reolink host."""
+        self._api.expire_session()
+
+        if not await self._api.get_host_data():
+            return False
+
+        if self._api.mac_address is None:
+            return False
+
+        enable_onvif = None
+        enable_rtmp = None
+        enable_rtsp = None
+
+        if not self._api.onvif_enabled:
+            _LOGGER.info(
+                "ONVIF is disabled on %s, trying to enable it", self._api.nvr_name
+            )
+            enable_onvif = True
+
+        if not self._api.rtmp_enabled and self._api.protocol == "rtmp":
+            _LOGGER.info(
+                "RTMP is disabled on %s, trying to enable it", self._api.nvr_name
+            )
+            enable_rtmp = True
+        elif not self._api.rtsp_enabled and self._api.protocol == "rtsp":
+            _LOGGER.info(
+                "RTSP is disabled on %s, trying to enable it", self._api.nvr_name
+            )
+            enable_rtsp = True
```

Comment:
Please log at debug as the user doesn't really need to know this unless something goes wrong

Reply:
done, thanks